files/ebs_volume.sh

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+until ping -c 1 google.com &> /dev/null; do
+  echo "waiting for outbound connectivity"
+  sleep 5
+done
+
+# Set common variables used throughout the script.
+imds_token=$( curl -Ss -H "X-aws-ec2-metadata-token-ttl-seconds: 300" -XPUT 169.254.169.254/latest/api/token )
+instance_id=$( curl -Ss -H "X-aws-ec2-metadata-token: $imds_token" 169.254.169.254/latest/meta-data/instance-id )
+availability_zone=$( curl -Ss -H "X-aws-ec2-metadata-token: $imds_token" 169.254.169.254/latest/meta-data/placement/availability-zone )
+volume_id=""
+
+# Search for an available EBS volume to attach to the instance. Wait one minute for a volume to become available,
+# if no volume is found - create new one, attach, format and mount the volume.
+
+for i in $(seq 1 6); do
+
+  volume_id=$(
+    aws --cli-connect-timeout 300 ec2 describe-volumes \
+      --filters "Name=status,Values=available" "Name=availability-zone,Values=$availability_zone" "Name=tag:Name,Values=${name}-graphdb-data" \
+      --query "Volumes[*].{ID:VolumeId}" \
+      --output text | \
+      sed '/^$/d'
+  )
+
+  if [ -z "${volume_id:-}" ]; then
+    echo 'ebs volume not yet available'
+    sleep 10
+  else
+    break
+  fi
+done
+
+if [ -z "${volume_id:-}" ]; then
+
+# If the volume type is GP3, then we need to provide a value for the "throughput" parameter.
+ if [ "${ebs_volume_type}" == "gp3"] || ["${ebs_volume_type}" == "gp2" ]; then
```

Comment:
Except for the `--throughput` option, both commands are identical. I suggest we define a variable before the if/else and set/reset its value and then use a single command, e.g.

```bash
throughput_option="--throughput ${ebs_volume_throughput}"
if [ "${ebs_volume_type}" != "gp3"] || ["${ebs_volume_type}" != "gp2" ]; then
    throughput_option=""
fi

volume_id=$(
        aws --cli-connect-timeout 300 ec2 create-volume \
          --availability-zone "$availability_zone" \
          --encrypted \
          --kms-key-id "${ebs_kms_key_arn}" \
          --volume-type "${ebs_volume_type}" \
          --size "${ebs_volume_size}" \
          --iops "${ebs_volume_iops}" \
          ${throughput_option:-} \
          --tag-specifications "ResourceType=volume,Tags=[{Key=Name,Value=${name}-graphdb-data}]" | \
          jq -r .VolumeId
)
```

note that we need `${throughput_option:-}` because of the `set -u` in the beginning.

Reply:
I have added the following code that will check if the ebs volume type is gp2 or gp3. If the value is gp2 or gp3, it will set a value for the throughput_option variable; if it is not, the variable will be an empty string.


`throughput_option=""
if [ "${ebs_volume_type}" = "gp3" ] || [ "${ebs_volume_type}" = "gp2" ]; then
    throughput_option="--throughput ${ebs_volume_throughput}"
else
    throughput_option=""
fi`